### PR TITLE
Institutions table

### PR DESCRIPTION
**Pull request by trombonekenny**
- Move `institution` access control out of hard-coded functions and into an `institutions` table
- Add tests for the `check_course_instance_access` sprocs
- Break some other hard-coded defaults out to config entries.
- Adds a viewing interface in `administratorOverhead` - unsure if web editing was needed

While this moves which institutions are checked in the access rule to be checked from a database table, it doesn't introduce any referential integrity between that table and the access rules. The access rule will fail if an institution is referenced that is not in the database.

### CODE REVIEW

File: lib/config.js [hunk 1]
```diff
@@ -80,6 +80,8 @@ config.questionRenderCacheType = 'none'; // One of none, redis, memory
 config.hasLti = false;
 config.ltiRedirectUrl = null;
 config.filesRoot = null;
+config.institutionDefault = 'UIUC';
+config.timezoneDefault = 'America/Chicago';
```
Comment (mwest1066) [inline lib/config.js]: I recommend deleting this (see other comment).

File: lib/config.js [hunk 2]
```diff
@@ -80,6 +80,8 @@ config.questionRenderCacheType = 'none'; // One of none, redis, memory
 config.hasLti = false;
 config.ltiRedirectUrl = null;
 config.filesRoot = null;
+config.institutionDefault = 'UIUC';
```
Comment (mwest1066) [inline lib/config.js]: I think we should set this to `'Default'` (see comment on the migration).

File: migrations/152_institutions__create.sql [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    institution TEXT NOT NULL,
+    uid_pattern TEXT
+);
+
+CREATE INDEX IF NOT EXISTS institutions_institution_idx ON institutions(institution);
+
+INSERT INTO institutions (id, institution, uid_pattern) VALUES
+    (1, 'UIUC', '%@illinois.edu'),
+    (2, 'ZJUI', '%@intl.zju.edu.cn'),
+    (3, 'gvsu.edu', '%@mail.gvsu.edu');
```
Comment (mwest1066) [inline migrations/152_institutions__create.sql]: I think we should just make this:
```
INSERT INTO institutions (id, institution) VALUES (1, 'Default');
```

File: database/tables/institutions.pg [hunk 4]
```diff
@@ -0,0 +1,11 @@
+columns
```
[the remark this answers is not in the document]
Comment (trombonekenny) [reply database/tables/institutions.pg]: I'm less inclined to change the terminology unless we're also changing the purpose. This keeps it consistent with access rules already in flight.

File: database/tables/institutions.pg [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+    id: bigint not null default nextval('institutions_id_seq'::regclass)
+    institution: text not null
```
[the remark this answers is not in the document]
Comment (trombonekenny) [reply database/tables/institutions.pg]: I liked `short_name` and `long_name` to mimic the course instance config.

File: migrations/152_institutions__create.sql (hunk 3 above)
Comment (trombonekenny) [reply migrations/152_institutions__create.sql]: Then the prairielearn.engr.illinois.edu will need some entries when this goes live. But I agree with the change.

File: schemas/schemas/serverConfig.json [hunk 6]
```diff
@@ -36,6 +36,14 @@
             "description": "Filename to use for server logging.",
             "type": "string"
         },
+        "institutionDefault": {
+            "description": "Default institution applied to courseInstance allowAccess",
+            "type": "string"
+        },
+        "timezoneDefault": {
```
[the remark this answers is not in the document]
Comment (trombonekenny) [reply schemas/schemas/serverConfig.json]: I wound up reverting configs for timezone and default institution.

File: database/tables/institutions.pg (hunk 4 above)
[the remark this answers is not in the document]
Comment (mwest1066) [reply database/tables/institutions.pg]: Sounds good to me.

**Pull request merged**